var/spack/repos/builtin/packages/cmake/package.py

```diff
@@ -69,6 +69,8 @@ class Cmake(Package):
     # Cannot build with Intel, should be fixed in 3.6.2
     # https://gitlab.kitware.com/cmake/cmake/issues/16226
     patch('intel-c-gnu11.patch', when='@3.6.0:3.6.1')
+    patch('ibm-xl.patch', when='%xl')
```

Comment:
They are for all versions, not only 3.5.2.  IMO, it should be applied to all cmake versions.